mautic/mautic
# Bigint unsigned ids

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | /
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8361
| BC breaks? | Kindof
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This is copy of https://github.com/mautic/mautic/pull/7081 which was merged and then reverted as it was considered a BC break. So Mautic 3 is a good release to get it in.

Heath wrote an amazing article about what happens when the autoincrement IDs runs out and how hard it is to fix it:

https://www.linkedin.com/pulse/so-you-hit-2147483647-heath-dutton-/

This PR will prevent that.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. It's hard to reproduce as you'd have to generate over 2.147.483.647 rows in one of the Mautic tables.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Make sure everything works as before. It should have no effect on existing Mautic instances but it should set the right column types for the new ones.

#### List backwards compatibility breaks:
1. One downside is that `bin/console doctrine:schema:update --force` will not work anymore as the migration is very difficult. It might be faster to create a fresh database and copy the data over.
